### DISCUSSION TITLE
User Photos Edit Asset like normal Assets

### DISCUSSION THREAD

**Discussion by davidhellmann** [craftcms/cms#8943] [Ideas]
Would be nice to have the possibility to rename Title and Filename for User Photos same as on other Asset Fields. (double click on it opens the Asset Details). 

Also it would be nice to add additional fields on that assets. In our case we need a Photo Credit Field. Would be great to have it directly on the asset and not as separate Field. Connect whats belongs to each other.

**Comment by brandonkelly** [comment]
User photos _are_ normal assets, and you can assign custom fields to them and edit their filename/field values, from the Assets section. Just not from the Edit User page.

**Reply by MoritzLost** [reply]
@brandonkelly We're also facing this issue at the moment, in a project where we make heavy use of users, both for frontend accounts and for a general people database (as contact persons, speaker in events, etc). We usually use a section with a custom assets field for the portrait, but with users, we're want to use the `photo` attribute. But the photo attribute is severely limited in multiple ways:

- It's not possible to pick an existing asset, you can only upload a file from your computer when you want to change the photo. This will lead to a lot of duplicate images.
- There's no way to edit the asset fields directly from the user form. You have to upload an image, then go to the asset library and find it there, which is very inconvenient.
- The photo attribute doesn't have the option to validate related fields that assets field have, so it's impossible to enforce specific fields (like the alt text) on the photo.
- In general, it's weird for authors that this single field looks and behaves differently than all other assets field. The fact that it's a native attribute and not a regular assets field is opaque to authors, so the distinction seems random. 

We're thinking about using the photo attribute only for internal use and placing a custom assets field for the frontend in the user field layout. But that redundancy is hard to explain to authors. Another option is to use User Visibility Conditions and hide the field for most users. But then you lose the ability to use a custom photo for the Control Panel, which is a shame.

Is there any reason the photo field behaves like this, other than inherited behaviour from previous Craft versions?

My ideal solution would be to keep the native photo property, but display it as a regular assets field with a limit of one.

**Comment by davidhellmann** [comment]
Yes, thats the point. Our client have to edit round about 300 Users and the imported Filenames for the User Photos are very unspeaking. Would be nice they can edit all the stuff directly on the user page. And sure, you're right with the custom fields and edit within the assets menu point. :)
